.github/workflows/windows-meson-msys2.yml

```diff
@@ -2,27 +2,166 @@ name: MSYS2 builds
+            mingw-w64-${{matrix.conf.toolchain}}${{matrix.conf.arch}}-fluidsynth
+            mingw-w64-${{matrix.conf.toolchain}}${{matrix.conf.arch}}-munt-mt32emu
+            mingw-w64-${{matrix.conf.toolchain}}${{matrix.conf.arch}}-libpng
+            mingw-w64-${{matrix.conf.toolchain}}${{matrix.conf.arch}}-opusfile
+            mingw-w64-${{matrix.conf.toolchain}}${{matrix.conf.arch}}-SDL2
+            mingw-w64-${{matrix.conf.toolchain}}${{matrix.conf.arch}}-SDL2_net
+            mingw-w64-${{matrix.conf.toolchain}}${{matrix.conf.arch}}-zlib
+      
+      - name:  Prepare compiler cache
+        id:    prep-ccache
+        run: |
+          mkdir -p "${CCACHE_DIR}"
+          echo "::set-output name=dir::C:\ccache"
+          echo "::set-output name=today::$(date -I)"
+          echo "::set-output name=yesterday::$(date --date=yesterday -I)"
+          echo "::set-output name=name_hash::$(echo '${{ matrix.conf.name }}' | shasum | cut -b-8)"
+
+      - uses:  actions/cache@v2
+        id:    cache-ccache
+        with:
+          path: ${{ steps.prep-ccache.outputs.dir }}
+          key:  ccache-msys2-${{ matrix.conf.sys }}-${{ steps.prep-ccache.outputs.name_hash }}-${{ steps.prep-ccache.outputs.today }}-2
+          restore-keys: |
+            ccache-msys2-${{ matrix.conf.sys }}-${{ steps.prep-ccache.outputs.name_hash }}-${{ steps.prep-ccache.outputs.yesterday }}-2
+
+      - name:  Cache subprojects
+        uses:  actions/cache@v2
+        with:
+          path: subprojects/packagecache
+          key:  subprojects-${{ hashFiles('subprojects/*.wrap') }}
+
+      - name: Log environment
+        run:  ./scripts/log-env.sh
+
+      - name: Setup build
+        run: |
+          meson setup ${{ matrix.conf.build_flags }} build
+      
+      - name: Build
+        run: |
+          set -xo pipefail
+          ninja -C build |& tee build.log
+          ccache -s
+      
+      - name: Run tests
+        if:   matrix.conf.run_tests
+        run:  meson test -C build --print-errorlogs
+
+      - name: Summarize warnings
+        if:   matrix.conf.run_tests != true
+        env:
+          MAX_WARNINGS: ${{ matrix.conf.max_warnings }}
+        run:  python ./scripts/count-warnings.py -lf build.log
+
+  build_msys2_release:
+    name: ${{ matrix.conf.name }} Release
+    runs-on: windows-latest
+    if: github.event_name != 'pull_request' || contains('dreamer,kcgen,ant-222,Wengier,shermp', github.actor) == false
+    
+    strategy:
+      matrix:
+        conf:
+          - name: GCC (MinGW) x86
+            arch: i686
+            sys: MINGW32
+          - name: GCC (MinGW) x86_64
+            arch: x86_64
+            sys: MINGW64
+    
+    defaults:
+      run:
+        shell: msys2 {0}
+   
+    steps:
+      - name: Disable git line ending conversion
+        shell: powershell
+        run: |
+          git config --global core.autocrlf false
+          git config --global core.eol lf
```

Comment:
This is a great find, and it would be even better if can we add it to the repo permanently, so users at home aren't tripped up with this trying to build with MSYS2.

Is it possible to add this as a `gitattributes` to the project, for the file-types that are impacted?

Reply:
It's a toss-up. Modifying `.gitattributes` is a big hammer to swing, and isn't actually required for MSYS2. Mainly because it's likely that the user will be using the MSYS2 provided version of git to clone the repo. I never had an issue of line endings in my local environment. Then again, I think I have the windows git installed to not touch line endings.

If you feel adding to `.gitattributes` is appropriate though, I can do that.

Note, even with this, invoking `count_warnings.py` via bash STILL failed, hence the fact I switched to invoking Python directly.

Reply:
Also don't know how modifying `.gitattributes` would affect MSVC users.

Reply:
Agreed; thanks @shermp . Let's avoid touching gitattributes for now.